sources/httpUtils.ts

```diff
@@ -6,25 +6,46 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
   if (process.env.COREPACK_ENABLE_NETWORK === `0`)
     throw new UsageError(`Network access disabled by the environment; can't reach ${url}`);
 
+  let redirects = parseInt(process.env.COREPACK_FETCH_MAX_REDIRECTS ?? ``);
+  if (isNaN(redirects) || redirects <= 0)
+    redirects = 3;
+
   const {default: https} = await import(`https`);
 
   const {ProxyAgent} = await import(`proxy-agent`);
 
   const proxyAgent = new ProxyAgent();
 
-  return new Promise<IncomingMessage>((resolve, reject) => {
-    const request = https.get(url, {...options, agent: proxyAgent}, response => {
-      const statusCode = response.statusCode;
-      if (statusCode != null && statusCode >= 200 && statusCode < 300)
-        return resolve(response);
+  const get = () => {
+    return new Promise<IncomingMessage>((resolve, reject) => {
+      const request = https.get(url, {...options, agent: proxyAgent});
 
-      return reject(new Error(`Server answered with HTTP ${statusCode} when performing the request to ${url}; for troubleshooting help, see https://github.com/nodejs/corepack#troubleshooting`));
-    });
+      request.on(`response`, response => {
+        const statusCode = response.statusCode!;
+        if (statusCode >= 200 && statusCode < 300)
+          return resolve(response);
+
+        if ((statusCode === 301 || statusCode === 302 || statusCode === 307 || statusCode === 308) && response.headers.location) {
```

Comment:
```suggestion
        if ([301, 302, 303, 307].includes(statusCode) && response.headers.location) {
```

This might be easier to read